File: Frontend/frontend_angular/src/app/services/mvtstk.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { MvtStkControllerService } from '../../gs-api/src/api/mvtStkController.service';
import { MvtStkRequestDto, MvtStkResponseDto } from '../../gs-api/src/model/models';

@Injectable({
  providedIn: 'root'
})
export class MvtStkService {

  constructor(private mvtStkController: MvtStkControllerService) { }

  findAll(): Observable<MvtStkResponseDto[]> {
    return this.mvtStkController.findAll3();
  }

  findById(id: number): Observable<MvtStkResponseDto> {
    return this.mvtStkController.findById3(id);
  }

  create(mvtStk: MvtStkRequestDto): Observable<MvtStkResponseDto> {
    return this.mvtStkController.save3(mvtStk);
  }

  update(id: number, mvtStk: MvtStkRequestDto): Observable<MvtStkResponseDto> {
    return this.mvtStkController.update3(id, mvtStk);
  }

  delete(id: number): Observable<any> {
    return this.mvtStkController.delete3(id);
  }
}

File: Backend/src/main/java/com/groupeO/gestiondestock/service/MvtStkService.java
package com.groupeO.gestiondestock.service;

import com.groupeO.gestiondestock.dto.MvtStkRequestDto;
import com.groupeO.gestiondestock.dto.MvtStkResponseDto;

import java.util.List;

public interface MvtStkService {
    MvtStkResponseDto save(MvtStkRequestDto mvtStkDto);
    MvtStkResponseDto findById(Integer id);
    List<MvtStkResponseDto> findAll();
    void delete(Integer id);
    MvtStkResponseDto update(Integer id, MvtStkRequestDto mvtStkDto);
}
